### ISSUE TITLE
Unable to Sync 

### ISSUE THREAD

**Issue by ehabhussein** [NethermindEth/nethermind#3606]
**Describe the bug**
Yesterday the Node (Full) i cant seem to discover any peers. i have also updated to the latest version but still doesnt work. I left it looking for peers for more than 8 hours.

I ran another instance on another machine to test. i can sync with no issues.

**To Reproduce**
1. $ /usr/share/nethermind/Nethermind.Runner

**Expected behavior**
Connect to nodes and sync

**Screenshots**
```
$ /usr/share/nethermind/Nethermind.Runner
2021-11-15 10-28-38.7324|Nethermind starting initialization.
2021-11-15 10-28-38.8238|Loading embedded plugins
2021-11-15 10-28-38.8240|  Found plugin type Nethermind.Consensus.AuRa.AuRaPlugin
2021-11-15 10-28-38.8241|  Found plugin type Nethermind.Consensus.Clique.CliquePlugin
2021-11-15 10-28-38.8241|  Found plugin type Nethermind.Consensus.Ethash.EthashPlugin
2021-11-15 10-28-38.8241|  Found plugin type Nethermind.Consensus.Ethash.NethDevPlugin
Resolved executing directory as /usr/share/nethermind.
2021-11-15 10-28-38.8475|Loading 6 assemblies from /usr/share/nethermind/plugins
2021-11-15 10-28-38.8478|Loading assembly Microsoft.Extensions.Diagnostics.HealthChecks
2021-11-15 10-28-38.8502|Loading assembly Nethermind.Api
2021-11-15 10-28-38.8526|Loading assembly Nethermind.Merge.Plugin
2021-11-15 10-28-38.8557|  Found plugin type Nethermind.Merge.Plugin
2021-11-15 10-28-38.8558|Loading assembly Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions
2021-11-15 10-28-38.8564|Loading assembly Nethermind.HealthChecks
2021-11-15 10-28-38.8575|  Found plugin type Nethermind.HealthChecks
2021-11-15 10-28-38.8575|Loading assembly Nethermind.Mev
2021-11-15 10-28-38.8621|  Found plugin type Nethermind.Mev
2021-11-15 10-28-39.0001|Loading standard NLog.config file from /usr/share/nethermind/NLog.config.
2021-11-15 10-28-39.3359|NLog.config loaded in 335ms.
2021-11-15 10-28-39.3373|Reading config file from /usr/share/nethermind/configs/mainnet.cfg
2021-11-15 10-28-39.7781|Incorrect config settings found:
ConfigType:JsonConfigFile|Category:AnalyticsConfig|Name:LogPublishedData
ConfigType:JsonConfigFile|Category:AnalyticsConfig|Name:PluginsEnabled
2021-11-15 10-28-39.7782|Configuration initialized.
2021-11-15 10:28:40.5147|Using http://ipv4.icanhazip.com to get external ip
2021-11-15 10:28:44.1765|Start block loaded from reorg boundary - 13616333 -
2021-11-15 10:28:44.2297|Numbers resolved, level = Max(13043999, 13616334), header = Max(13043999, 13616334), body = Max(0, 13616332)
2021-11-15 10:28:44.2307|Block tree initialized, last processed is 0, best queued is 13616334, best known is 13616334, lowest inserted header 1, body 11052190
2021-11-15 10:28:44.2429|Initializing 7 plugins
2021-11-15 10:28:44.2429|  AuRa by Nethermind
2021-11-15 10:28:44.2429|  AuRa by Nethermind initialized in 0ms
2021-11-15 10-28-38.8241|  Found plugin type Nethermind.Consensus.Clique.CliquePlugin
2021-11-15 10-28-38.8241|  Found plugin type Nethermind.Consensus.Ethash.EthashPlugin
2021-11-15 10-28-38.8241|  Found plugin type Nethermind.Consensus.Ethash.NethDevPlugin
Resolved executing directory as /usr/share/nethermind.
2021-11-15 10-28-38.8475|Loading 6 assemblies from /usr/share/nethermind/plugins
2021-11-15 10-28-38.8478|Loading assembly Microsoft.Extensions.Diagnostics.HealthChecks
2021-11-15 10-28-38.8502|Loading assembly Nethermind.Api
2021-11-15 10-28-38.8526|Loading assembly Nethermind.Merge.Plugin
2021-11-15 10-28-38.8557|  Found plugin type Nethermind.Merge.Plugin
2021-11-15 10-28-38.8558|Loading assembly Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions
2021-11-15 10-28-38.8564|Loading assembly Nethermind.HealthChecks
2021-11-15 10-28-38.8575|  Found plugin type Nethermind.HealthChecks
2021-11-15 10-28-38.8575|Loading assembly Nethermind.Mev
2021-11-15 10-28-38.8621|  Found plugin type Nethermind.Mev
2021-11-15 10-28-39.0001|Loading standard NLog.config file from /usr/share/nethermind/NLog.config.
2021-11-15 10-28-39.3359|NLog.config loaded in 335ms.
2021-11-15 10-28-39.3373|Reading config file from /usr/share/nethermind/configs/mainnet.cfg
2021-11-15 10-28-39.7781|Incorrect config settings found:
ConfigType:JsonConfigFile|Category:AnalyticsConfig|Name:LogPublishedData
ConfigType:JsonConfigFile|Category:AnalyticsConfig|Name:PluginsEnabled
2021-11-15 10-28-39.7782|Configuration initialized.
2021-11-15 10:28:40.5147|Using http://ipv4.icanhazip.com to get external ip
2021-11-15 10:28:44.1765|Start block loaded from reorg boundary - 13616333 -
2021-11-15 10:28:44.2297|Numbers resolved, level = Max(13043999, 13616334), header = Max(13043999, 13616334), body = Max(0, 13616332)
2021-11-15 10:28:44.2307|Block tree initialized, last processed is 0, best queued is 13616334, best known is 13616334, lowest inserted header 1, body 11052190
2021-11-15 10:28:44.2429|Initializing 7 plugins
2021-11-15 10:28:44.2429|  AuRa by Nethermind
2021-11-15 10:28:44.2429|  AuRa by Nethermind initialized in 0ms
2021-11-15 10:28:44.2429|  Clique by Nethermind
2021-11-15 10:28:44.2429|  Clique by Nethermind initialized in 0ms
2021-11-15 10:28:44.2429|  Ethash by Nethermind
2021-11-15 10:28:44.2519|  Ethash by Nethermind initialized in 7ms
2021-11-15 10:28:44.2519|  NethDev by Nethermind
2021-11-15 10:28:44.2519|  NethDev by Nethermind initialized in 0ms
2021-11-15 10:28:44.2519|  Merge by Nethermind
2021-11-15 10:28:44.2519|  Merge by Nethermind initialized in 0ms
2021-11-15 10:28:44.2519|  HealthChecks by Nethermind
2021-11-15 10:28:44.2519|  HealthChecks by Nethermind initialized in 0ms
2021-11-15 10:28:44.2519|  MEV by Nethermind
2021-11-15 10:28:44.2548|  MEV by Nethermind initialized in 0ms
2021-11-15 10:28:44.6117|Loaded 0 static nodes from file: /usr/share/nethermind/Data/static-nodes.json
2021-11-15 10:28:44.6715|Using pivot 13044000 and barrier 11052984 in bodies sync
2021-11-15 10:28:44.6715|Using pivot 13044000 and barrier 11052984 in receipts sync
2021-11-15 10:28:44.6805|HeadersSyncDispatcher has finished work.
2021-11-15 10:28:44.6805|Fast blocks headers task completed.
2021-11-15 10:28:44.6857|Now syncing nodes starting from root of block 0
2021-11-15 10:28:44.7089|No block tree levels to review for fixes. All fine.
2021-11-15 10:28:44.7201|Numbers resolved, level = Max(13043999, 13616334), header = Max(13043999, 13616334), body = Max(0, 13616332)
2021-11-15 10:28:44.8200|Started monitoring for the group: nethermind, instance: MainnetFast
2021-11-15 10:28:44.9163|Health RPC Module has been enabled
2021-11-15 10:28:44.9163|Skipping Flashbots RPC plugin
2021-11-15 10:28:45.3997|Peers | with known best block: 0 | all: 0 |
2021-11-15 10:28:45.3997|Waiting for peers... 1s
2021-11-15 10:28:45.4052|Sync peers - Initialized: 0 | All: 0 | Max: 100
2021-11-15 10:28:45.7813|Connected to 8 bootnodes, 8 trusted/persisted nodes
2021-11-15 10:28:46.4049|Waiting for peers... 2s
2021-11-15 10:28:46.5837|
======================== Nethermind initialization completed ========================
Version      : 1.11.7-0-75f034a08-20211115/X64-Linux/5.0.11
This node    : enode://3xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx@xxxxxxxxxxx:30305
RPC modules  : Admin, Debug, Eth, Evm, Net, Parity, Personal, Proof, Subscribe, Trace, TxPool, Web3
Node address : xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx(do not use as an account)
Mem est tx   :    115MB
Mem est DB   :  40993MB
JSON RPC     : http://0.0.0.0:8665
Genesis hash : 0xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
External IP  : xxxxxxxxxxx
Ethereum     : tcp://xxxxxxxxx:30305
Discovery    : udp://xxxxxxxxxx:30305
Chainspec    : chainspec/foundation.json
Chain ID     : Mainnet
Chain head   : 0
=====================================================================================
2021-11-15 10:28:47.4041|Waiting for peers... 3s
2021-11-15 10:28:48.4046|Waiting for peers... 4s
2021-11-15 10:28:49.4031|Waiting for peers... 5s
2021-11-15 10:28:50.4047|Waiting for peers... 6s
2021-11-15 10:28:51.4040|Waiting for peers... 7s
2021-11-15 10:30:14.4036|Waiting for peers... 30s
2021-11-15 10:30:15.4048|Peers | with known best block: 0 | all: 0 |
2021-11-15 10:30:15.4048|Waiting for peers... 31s
2021-11-15 10:30:16.4048|Waiting for peers... 32s
2021-11-15 10:30:17.4045|Waiting for peers... 33s
2021-11-15 10:30:18.4036|Waiting for peers... 34s
2021-11-15 10:30:19.4042|Waiting for peers... 35s
2021-11-15 10:30:20.4042|Waiting for peers... 36s
2021-11-15 10:30:21.4045|Waiting for peers... 37s
2021-11-15 10:30:22.4033|Waiting for peers... 38s
2021-11-15 10:30:23.4039|Waiting for peers... 39s
2021-11-15 10:30:24.4047|Waiting for peers... 40s
2021-11-15 10:30:25.4044|Waiting for peers... 41s
2021-11-15 10:30:26.4031|Waiting for peers... 42s
2021-11-15 10:30:27.4028|Waiting for peers... 43s
2021-11-15 10:30:28.4060|Waiting for peers... 44s
2021-11-15 10:30:29.4036|Waiting for peers... 45s
2021-11-15 10:30:30.4034|Peers | with known best block: 0 | all: 0 |
2021-11-15 10:30:30.4034|Waiting for peers... 46s
2021-11-15 10:30:31.4041|Waiting for peers... 47s
2021-11-15 10:30:32.4044|Waiting for peers... 48s
2021-11-15 10:30:33.4034|Waiting for peers... 49s
2021-11-15 10:30:34.4034|Waiting for peers... 50s
2021-11-15 10:30:35.4034|Waiting for peers... 51s
2021-11-15 10:30:36.4050|Waiting for peers... 52s
2021-11-15 10:30:37.4054|Waiting for peers... 53s
2021-11-15 10:30:38.4047|Waiting for peers... 54s
2021-11-15 10:30:39.4044|Waiting for peers... 55s
2021-11-15 10:30:40.4035|Waiting for peers... 56s
2021-11-15 10:30:41.4045|Waiting for peers... 57s
2021-11-15 10:30:42.4043|Waiting for peers... 58s
2021-11-15 10:30:43.4051|Waiting for peers... 59s
2021-11-15 10:30:44.4039|Waiting for peers... 0s
2021-11-15 10:30:45.4053|Peers | with known best block: 0 | all: 0 |
2021-11-15 10:30:45.4053|Waiting for peers... 1s
2021-11-15 10:30:45.4053|Sync peers - Initialized: 0 | All: 0 | Max: 100
2021-11-15 10:30:46.4032|Waiting for peers... 2s
2021-11-15 10:30:47.4055|Waiting for peers... 3s
2021-11-15 10:30:48.4057|Waiting for peers... 4s
2021-11-15 10:30:49.4044|Waiting for peers... 5s
2021-11-15 10:30:50.4031|Waiting for peers... 6s
2021-11-15 10:30:51.4046|Waiting for peers... 7s
2021-11-15 10:30:52.4037|Waiting for peers... 8s
2021-11-15 10:30:53.4038|Waiting for peers... 9s
2021-11-15 10:30:54.4030|Waiting for peers... 10s
```

**Desktop (please complete the following information):**
 - ubuntu 20.04.3 LTS

**Comment by LukaszRozmej** [comment]
Most of the time on a popular network like mainnet this is caused by network configuration issue, can you check port 30305 is open and routed correctly, also if your host IP is correctly recognized? Your JSON RPC address 0.0.0.0 might signal its not. You can try set ExternalIP in network config: https://docs.nethermind.io/nethermind/ethereum-client/configuration/network

Otherwise @matilote @1swaraj can you advise?

**Comment by ehabhussein** [comment]
@LukaszRozmej I have tried all that , the issue is still the same.

**Comment by ehabhussein** [comment]
![image](https://user-images.githubusercontent.com/5464055/141754736-d9027bbe-7159-4347-8aae-76f01af07560.png)

Ive added Some grafana screenshot .

**Comment by ehabhussein** [comment]
Enabled Trace for the node, below was the exceptions that were caught:

``` Csharp
2021-11-15 13:44:43.0193|Bonded with 172xxxx5.145
2021-11-15 13:44:43.0193|Received message: Pong
2021-11-15 13:44:43.0193|Error in communication with [Node|10xxx.7:30303|Geth/v1.10.11-stable-7231b3ef/linux-amd64/go1.17.2|eth66]: System.InvalidOperationException: Cannot initialize eth66 without the head block set
   at Nethermind.Network.P2P.Subprotocols.Eth.V62.Eth62ProtocolHandler.Init()
   at Nethermind.Network.ProtocolsManager.InitProtocol(ISession session, String protocolCode, Int32 version, Boolean addCapabilities)
   at Nethermind.Network.P2P.P2PProtocolHandler.HandleMessage(Packet msg)
   at Nethermind.Network.P2P.Session.ReceiveMessage(ZeroPacket zeroPacket)
   at DotNetty.Transport.Channels.SimpleChannelInboundHandler`1.ChannelRead(IChannelHandlerContext ctx, Object msg)
   at DotNetty.Transport.Channels.AbstractChannelHandlerContext.InvokeChannelRead(Object msg)
```

``` Csharp
2021-11-15 13:50:04.6750|Sending message: Type: Pong, FarAddress: 52.xxx.120.184:30303, FarPublicKey: empty, ExpirationTime: 1636977024, PingMdc: 3be188c69xxxxxxxxxxxxxx6b3c8f13f1dd99e2d6df5f5
2021-11-15 13:50:04.6750||NetworkTrace| [Session|Out|Disconnected|30305->171.xxx.224.24:30303] channel disconnected
2021-11-15 13:50:04.6750|[Session|Out|Disconnected|30305->171.100.224.24:30303] already disconnected ClientQuitting Remote
2021-11-15 13:50:04.6750|Channel Unregistered
2021-11-15 13:50:04.6750|Handshake with 0x0840d9a23fa3ec7dd0d188c7f429b5xxxxxxxxx482f50301820a3dec4178cdba4f60992adf @  finished. Removing NettyHandshakeHandler from the pipeline
2021-11-15 13:50:04.6750|Error while disconnecting on channel on [Session|Out|Disconnected|30305->171.xxx.224.24:30303] : System.AggregateException: One or more errors occurred. (Transport endpoint is not connected)
 ---> System.Net.Sockets.SocketException (107): Transport endpoint is not connected
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.Shutdown(SocketShutdown how)
   at DotNetty.Transport.Channels.Sockets.TcpSocketChannel.DoClose()
   at DotNetty.Transport.Channels.AbstractChannel.AbstractUnsafe.DoClose0(TaskCompletionSource promise)
   --- End of inner exception stack trace ---

```

``` Csharp
2021-11-15 13:50:06.6584|Error when sending a discovery message Msg: Type: FindNode, FarAddress: 51xxxxx.17:30303, FarPublicKey: empty, ExpirationTime: 1636977026, SearchedNodeId: f8ae76a35c202xxxxxxxxxxxxxxe9404a34b22747ab8ce355efb188afe5bxxxxxxxc3cea6d45a20d40185 ,Exp: System.AggregateException: One or more errors occurred. (I/O error occurred.)
 ---> DotNetty.Transport.Channels.ClosedChannelException: I/O error occurred.
   --- End of inner exception stack trace ---
2021-11-15 13:50:06.6590|Error when sending a discovery message Msg: Type: FindNode, FarAddress: 16xxxxx46:30303, FarPublicKey: empty, ExpirationTime: 1636977026, SearchedNodeId: f8ae76a35c202exxxxx2747ab8ce355efb188afe5b177d5aa9c8d2848134cbc3cea6d45a20d40185 ,Exp: System.AggregateException: One or more errors occurred. (I/O error occurred.)
 ---> DotNetty.Transport.Channels.ClosedChannelException: I/O error occurred.
   --- End of inner exception stack trace ---
2021-11-15 13:50:06.6590|Error when sending a discovery message Msg: Type: FindNode, FarAddress: 131.1xxxxx.9.166:30311, FarPublicKey: empty, ExpirationTime: 1636977026, SearchedNodeId: f8ae76a35c202e5c15xxxxxefb188afe5b177d5axxxxx48134cbc3cea6d45a20d40185 ,Exp: System.AggregateException: One or more errors occurred. (I/O error occurred.)
 ---> DotNetty.Transport.Channels.ClosedChannelException: I/O error occurred.
   --- End of inner exception stack trace ---
```

**Comment by LukaszRozmej** [comment]
@ehabhussein have you been able to solve the issue?

**Comment by ehabhussein** [comment]
I haven't . I have opted with a redeployment of the node from scratch ...

**Comment by LukaszRozmej** [comment]
Ah sorry, didn't come to my mind that a debug_resetHead might have recovered the node.
Also added some recovery for this corruption for BlockTree, hard to find  a root cause other than debug_resetHead itself,

**Comment by DLTS-FD** [comment]
Hello @LukaszRozmej is there a tutorial how to use `debug_resetHead` ? I am using nethermind docker-images and I am getting

`execution-client | 2023-06-30 12:55:19.4128|Detected corrupted block tree data (28652008 < 28652009) (possibly due to an unexpected shutdown). Attempting to fix by moving head backwards.`

What can I add to the docker-compose file in order to start the syncing from an earler block head?

**Comment by LukaszRozmej** [comment]
> Hello @LukaszRozmej is there a tutorial how to use `debug_resetHead` ? I am using nethermind docker-images and I am getting
> 
> `execution-client | 2023-06-30 12:55:19.4128|Detected corrupted block tree data (28652008 < 28652009) (possibly due to an unexpected shutdown). Attempting to fix by moving head backwards.`
> 
> What can I add to the docker-compose file in order to start the syncing from an earler block head?

Does it manage to recover? Is your node an archive node? (If not to both questions it is probably easier to resync)

**Comment by DLTS-FD** [comment]
Hey @LukaszRozmej thanks for coming back. Both answers are no, and so we decided to do a full re-sync last week. 

We suspect we ran into an incompatible block by using a combination of an updated nethermind client with a still not yet updated lighthouse node. After the resync we are back operationable. 

Just as a general hint, I could not find a way to use the setHead method if the node crashes and restarts in a docker image. Ideally a flag for node startup which could be used in the ' docker-compose.yaml' file would be nice.
